com/bl/ttt/tictactoe.py

```diff
@@ -0,0 +1,62 @@
+import re
+
+
+class Tictactoe:
+    """ Cell coordinates are:
+    0	1	2
+    3	4	5
+    6	7	8
+    """
+    def __init__(self):
+        self.cells = [' ', ' ', ' ', ' ', ' ', ' ', ' ', ' ', ' ']
+
+    def is_valid_move(self, player_id, move):
```

Comment:
I think this should be 
`def is_valid_move(self, move):`

Reply:
Yes2, also incomplete refactoring.

Reply:
Can I suggest you refactor this function assuming that `TictactoeMove` has a `cell_chosen` attribute it will make things a lot clearer.